iroh-net/src/lib.rs

```diff
@@ -103,8 +112,72 @@
 //! up the [`NodeId`] in the DNS system to find the addressing details.  This enables
 //! connecting using only the [`NodeId`] which is often more convenient and resilient.
 //!
+//! See [the discovery module] for more details.
+//!
+//!
+//! # Examples
+//!
+//! The central struct is the [`Endpoint`], which allows you to connect to other nodes:
+//!
+//! ```no_run
```

Comment:
does `no_run` still check if this compiles? otherwise I am very worried about these getting out of date

Reply:
Yep, `no_run` checks it compiles but does not try to run it.  Seemed like the right compromise here, since building a fully workable example was so much extra noise.  I guess I could hide all that noise to still show it like this, but the hiding is a pretty horrible experience to write as well...